src/TabBar.tsx

```diff
@@ -299,18 +310,23 @@ export default class TabBar<T extends Route> extends React.Component<
       contentContainerStyle,
       style,
       indicatorContainerStyle,
+      tabBarSpacing,
     } = this.props;
     const { layout, tabWidths } = this.state;
     const { routes } = navigationState;
 
     const isWidthDynamic = this.getFlattenedTabWidth(tabStyle) === 'auto';
     const tabBarWidth = this.getTabBarWidth(this.props, this.state);
+    const separatorsWidth = Math.max(0, routes.length - 1) * tabBarSpacing;
+
     const tabBarWidthPercent = `${routes.length * 40}%`;
     const translateX = this.getTranslateX(
       this.scrollAmount,
       this.getMaxScrollDistance(tabBarWidth, layout.width)
     );
 
+    const separatorPercent = (separatorsWidth / tabBarWidth) * 100;
```

Comment:
You can move this near the declaration of `separatorsWidth` to have related calculations one after the other